From 18dbdabe14c7972313e4e13b3a431fa7efd7cdfb Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Wed, 28 Mar 2018 19:30:26 +0200
Subject: [PATCH 4/6] fs: sdfat: Don't flush plug so often in bdev_readahead

* The plug will be flushed when we call finish, and this
  breaks compilation when this driver is built as a module

Change-Id: I9e774bada1ddbb451d43aa30491369cd7aed107e
---
 fs/sdfat/blkdev.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/fs/sdfat/blkdev.c b/fs/sdfat/blkdev.c
index 264c670df0f..5082435b07e 100644
--- a/fs/sdfat/blkdev.c
+++ b/fs/sdfat/blkdev.c
@@ -109,7 +109,6 @@ s32 bdev_check_bdi_valid(struct super_block *sb)
 s32 bdev_readahead(struct super_block *sb, u64 secno, u64 num_secs)
 {
 	FS_INFO_T *fsi = &(SDFAT_SB(sb)->fsi);
-	u32 sects_per_page = (PAGE_SIZE >> sb->s_blocksize_bits);
 	struct blk_plug plug;
 	u64 i;
 
@@ -117,11 +116,8 @@ s32 bdev_readahead(struct super_block *sb, u64 secno, u64 num_secs)
 		return -EIO;
 
 	blk_start_plug(&plug);
-	for (i = 0; i < num_secs; i++) {
-		if (i && !(i & (sects_per_page - 1)))
-			blk_flush_plug(current);
+	for (i = 0; i < num_secs; i++)
 		sb_breadahead(sb, (sector_t)(secno + i));
-	}
 	blk_finish_plug(&plug);
 
 	return 0;
-- 
2.15.1

